src/Compilers/Core/CodeAnalysisTest/MetadataReferences/ModuleMetadataTests.cs

```diff
@@ -61,8 +61,7 @@ public unsafe void CreateFromImage()
             Assert.Throws<ArgumentOutOfRangeException>(() => { fixed (byte* ptr = new byte[] { 1, 2, 3 }) ModuleMetadata.CreateFromImage((IntPtr)ptr, -1); });
 
             Assert.Throws<ArgumentNullException>(() => ModuleMetadata.CreateFromImage(default(ImmutableArray<byte>)));
-            Assert.Throws<ArgumentNullException>(() => ModuleMetadata.CreateFromImage(default(IEnumerable<byte>)));
-            Assert.Throws<ArgumentNullException>(() => ModuleMetadata.CreateFromImage(default(byte[])));
+            Assert.Throws<ArgumentNullException>(() => ModuleMetadata.CreateFromImage(null));
```

Comment:
The two cases could be restored with casts.

Reply:
The two cases call the same method.

Reply:
Um... this seems questionable. The test is useful to me, as it shows that I can call with a `null` `byte[]`. Yes, that's only the result of a conversion, but it still captures useful information about user experience.
I do not think we should remove the extra call, let alone automatically.

Reply:
Restored the tests in #42546